src/main/java/ch/njol/skript/expressions/ExprJoinSplit.java

```diff
@@ -35,43 +12,48 @@
 import ch.njol.skript.lang.util.SimpleExpression;
 import ch.njol.util.Kleenean;
 import ch.njol.util.StringUtils;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.UnknownNullability;
+
+import java.util.regex.Pattern;
 
-/**
- * @author Peter Güttinger
- */
 @Name("Join & Split")
 @Description("Joins several texts with a common delimiter (e.g. \", \"), or splits a text into multiple texts at a given delimiter.")
 @Examples({
-		"message \"Online players: %join all players with \"\" | \"\"%\" # %all players% would use the default \"x, y, and z\"",
-		"set {_s::*} to the string argument split at \",\""
+	"message \"Online players: %join all players' names with \"\" | \"\"%\" # %all players% would use the default \"x, y, and z\"",
+	"set {_s::*} to the string argument split at \",\""
 })
-@Since("2.1, 2.5.2 (regex support), 2.7 (case sensitivity)")
+@Since("2.1, 2.5.2 (regex support), 2.7 (case sensitivity), INSERT VERSION (without trailing string)")
 public class ExprJoinSplit extends SimpleExpression<String> {
 
 	static {
 		Skript.registerExpression(ExprJoinSplit.class, String.class, ExpressionType.COMBINED,
 			"(concat[enate]|join) %strings% [(with|using|by) [[the] delimiter] %-string%]",
-			"split %string% (at|using|by) [[the] delimiter] %string% [case:with case sensitivity]",
-			"%string% split (at|using|by) [[the] delimiter] %string% [case:with case sensitivity]",
-			"regex split %string% (at|using|by) [[the] delimiter] %string%",
-			"regex %string% split (at|using|by) [[the] delimiter] %string%");
+			"split %string% (at|using|by) [[the] delimiter] %string% [case:with case sensitivity] [trailing:without [the] trailing [empty] string]",
+			"%string% split (at|using|by) [[the] delimiter] %string% [case:with case sensitivity] [trailing:without [the] trailing [empty] string]",
+			"regex split %string% (at|using|by) [[the] delimiter] %string% [trailing:without [the] trailing [empty] string]",
+			"regex %string% split (at|using|by) [[the] delimiter] %string% [trailing:without [the] trailing [empty] string]");
```

Comment:
it might be good to allow `text` in place of `string` as `text` is commonly used in other places